core/src/main/java/nl/nn/adapterframework/stream/Message.java

```diff
@@ -495,120 +501,136 @@ public long size() {
-		if (!requiresStream()) {
+	public <O extends OutputStream> O captureStream(int maxSize, O outputStream) {
+		if (!requiresStream() || !isBinary()) {
 			return null;
 		}
 		log.debug("creating capture of "+ClassUtils.nameOf(request));
-		if (isBinary()) {
-			OutputStream stream = new WriterOutputStream(writer, StreamUtil.DEFAULT_INPUT_STREAM_ENCODING);
-			try {
-				request = new FilterInputStream(asInputStream()) {
-					
-					int pos;
-					
-					@Override
-					public int read() throws IOException {
-						if (log.isTraceEnabled()) log.trace("FilterInputStream.read");
-						int result = super.read();
-						if (result>=0 && pos++<maxSize) {
-							stream.write((byte)result);
-							if (log.isTraceEnabled()) log.trace("captured byte ["+result+"]");
-						}
-						return result;
+		try {
+			request = new FilterInputStream(asInputStream()) {
+				
+				int pos;
+				
+				@Override
+				public int read() throws IOException {
+					if (log.isTraceEnabled()) log.trace("FilterInputStream.read");
+					int result = super.read();
+					if (result>=0 && pos++<maxSize) {
+						outputStream.write((byte)result);
+						if (log.isTraceEnabled()) log.trace("captured byte ["+result+"]");
 					}
-
-					@Override
-					public int read(byte[] b, int off, int len) throws IOException {
-						if (log.isTraceEnabled()) log.trace("FilterInputStream.read");
-						int result = super.read(b, off, len);
-						if (result>=0 && pos<maxSize) {
-							pos+=result;
-							stream.write(b, off, result);
-							if (log.isTraceEnabled()) log.trace("captured ["+result+"] bytes");
-						}
-						return result;
+					return result;
+				}
+
+				@Override
+				public int read(byte[] b, int off, int len) throws IOException {
+					if (log.isTraceEnabled()) log.trace("FilterInputStream.read");
+					int result = super.read(b, off, len);
+					if (result>=0 && pos<maxSize) {
+						pos+=result;
+						outputStream.write(b, off, result);
+						if (log.isTraceEnabled()) log.trace("captured ["+result+"] bytes");
 					}
-
-					@Override
-					public void close() throws IOException {
-						if (log.isTraceEnabled()) log.trace("FilterInputStream.close");
-						if (available()>0) {
-							if (log.isTraceEnabled()) log.trace("bytes available at close");
-							stream.write("(--MoreBytesAvailable--)[".getBytes());
-							read(new byte[1000]);
-							stream.write("]".getBytes());
-							if (read()>0) {
-								stream.write("...".getBytes());
-							}
-						}
-						super.close();
-						stream.close();
+					return result;
+				}
+
+				@Override
+				public void close() throws IOException {
+					if (log.isTraceEnabled()) log.trace("FilterInputStream.close");
+					if (available()>0) {
+						// Make the bytes available for debugger even when the stream was not used (might be because the
+						// pipe or sender that normally consumes the stream is stubbed by the debugger)
+						int len = read(new byte[maxSize]);
```

Comment:
Hier lees je alleen nog bytes, maar je kan ze niet zien in de debugger, want je schrijft ze niet naar de output.

Reply:
De read(new byte[maxSize]) roept de read(byte[] b, int off, int len) aan als het goed is die aantal regels boven de close() methode overschreven is. Ik heb dit ook getest met de Ladybug, kan het nog even dubbelchecken.

Reply:
hm ja, natuurlijk. sorry.